src/Microsoft.AspNet.Security.OAuth/OAuthAuthenticationHandler.cs

```diff
@@ -52,26 +52,21 @@ public OAuthAuthenticationHandler(HttpClient backchannel, ILogger logger)
 
             var context = new OAuthReturnEndpointContext(Context, ticket)
             {
-                SignInAsAuthenticationType = Options.SignInAsAuthenticationType,
+                SignInAsAuthenticationScheme = Options.SignInAsAuthenticationScheme,
                 RedirectUri = ticket.Properties.RedirectUri,
             };
             ticket.Properties.RedirectUri = null;
 
             await Options.Notifications.ReturnEndpoint(context);
 
-            if (context.SignInAsAuthenticationType != null && context.Identity != null)
+            if (context.SignInAsAuthenticationScheme != null && context.Principal != null)
```

Comment:
Ok sounds like we should throw/blow up if SignInScheme is null or empty early on then?